### ISSUE TITLE
repeat array not working with length(counts) more than array's dimensions

### ISSUE THREAD

**Issue by johnnychen94** [JuliaLang/julia#29614]
For simplicity, I remove part of results

For 1-d `Vector`, this works
```julia
julia> a = ones(2)
2-element Array{Float64,1}:

julia> repeat(a,1,2,3)
2×2×3 Array{Float64,3}:
```

For `Array` more than 1-d, this doesn't
```julia
julia> a = ones(2,2)
2×2 Array{Float64,2}:

julia> repeat(a,1,2,3) # I think this should work
ERROR: BoundsError: attempt to access 2×4×3 Array{Float64,3} at index [Base.OneTo(2), Base.OneTo(2)]
```

But this does,

```julia
julia> a = ones(2,2)
2×2 Array{Float64,2}:

julia> repeat(a,1,1,1)
2×2×1 Array{Float64,3}:

```

A temporary solution to repeat in higher dimension is to reshape first
```julia
julia> a = ones(2,2)
2×2 Array{Float64,2}:

julia> repeat(repeat(a,1,1,1),1,2,3)
2×4×3 Array{Float64,3}:

```
Platform Info:
  OS: Linux (x86_64-redhat-linux)
  CPU: Intel(R) Core(TM) i7-6800K CPU @ 3.40GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.0 (ORCJIT, broadwell)
```

**Comment by nalimilan** [comment]
I'd say this qualifies as a feature request. The fact that non-existent dimensions are accepted for vectors isn't documented AFAICT.

**Comment by johnnychen94** [comment]
fixed by #29626
